task/report_diskusage.go

```diff
@@ -0,0 +1,33 @@
+package task
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/emicklei/go-selfdiagnose"
+)
+
+const (
+	B  = 1
+	KB = 1024 * B
+	MB = 1024 * KB
+	GB = 1024 * MB
+)
+
+type ReportDiskusage struct {
+	Path string
+}
+
+func (r ReportDiskusage) Run(ctx *selfdiagnose.Context, result *selfdiagnose.Result) {
+	fs := syscall.Statfs_t{}
+	err := syscall.Statfs(r.Path, &fs)
+	if err != nil {
+		return
```

Comment:
you could put the error message in the result reason here and set the Passed to false explicitly